### PR TITLE
Malware detection script

### PR DESCRIPTION
**Pull request by mira-miracoli**
mostly documented in the code

todo:

- [ ] update tasks
- [ ] set env in bashrc
- [ ] add venv creation to ansible taks
- [ ] use venv and correct flags in cron

### CODE REVIEW

File: files/galaxy/mining-checksums/checksums.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+# width=32 poly=0x04c11db7 init=0x00000000 refin=false
+# refout=false xorout=0xffffffff check=0x765e7680
+# residue=0xc704dd7b name="CRC-32/CKSUM"
+# can be achieved using this bash command:
+# gzip -1 -c /path/to/file | tail -c8 | hexdump -n4 -e '"%u"'
+miner:
+  sse:
+    "2.1":
+      severity: high
+      description: ""
```
Comment (bgruening) [inline files/galaxy/mining-checksums/checksums.yml:22]: Would be nice to add descriptions.

File: roles/usegalaxy-eu.find-miners/templates/find_miners.py [hunk 2]
```diff
@@ -0,0 +1,521 @@
...
+                        version:
+                            severity: [high, medium, low]
+                            description: "optional info"
+                            checksums:
+                                crc32: <checksum crc32, gzip algorithm, integer representation>
+                                sha1: <checksum sha1, hex representation>
+            WARNING:
+            Be careful with how you generate the CRC32 hashes:
+            There are multiple algorithms, this script is using the one specified by IEEE 802.3
+            You should get this when using the gzip command on POSIX systems
```
Comment (bgruening) [inline roles/usegalaxy-eu.find-miners/templates/find_miners.py:51]: Do you have an example CMD here? I guess we want to make contributions to this library easy.

**Review by bgruening — COMMENTED** [review]
Maybe the script should be named malware_detection and not miners? Its more general, isn't it?

File: roles/usegalaxy-eu.find-miners/templates/find_miners.py (hunk 2 above)
Comment (mira-miracoli) [reply roles/usegalaxy-eu.find-miners/templates/find_miners.py:51]: yes I should add it. my current solution is only in the yaml file and reads not very smoothly.
`gzip -1 -c /path/to/file | tail -c8 | hexdump -n4 -e '"%u"'`

**Comment by mira-miracoli** [comment]
I think it could deserve it's own repo/ansible role. That would make it easier for others to use it.

**Pull request closed without merging**